QuizApp/Quizzes/QuizzesViewController.swift

```diff
@@ -85,41 +81,24 @@ class QuizzesViewController: UIViewController {
         tableView.dataSource = self
         tableView.delegate = self
         
-        view.addSubview(showButton)
         view.addSubview(tableView)
         view.addSubview(fact)
         view.addSubview(nbaQuestion)
         view.bringSubviewToFront(tableView)
     }
     
-    @objc
-    func customAction() {
-        tableView.isHidden = false
-        quizzesUseCase.fetchQuizzes()  {[weak self] quizzes in
-            DispatchQueue.main.async {
-                guard let self = self else { return }
-                self.quizzes = quizzes.quizzes
-                self.quizzesGroupedByCategory = self.groupByCategory(quizzesList: self.quizzes)
-                self.tableView.reloadData()
-            
-                let count = self.quizzes.map{$0.questions}.flatMap{$0}.filter{$0.question.contains("NBA")}.count
-            
-                self.fact.text = "Fun Fact"
-                self.nbaQuestion.text = "There are \(count) questions that contain the word \"NBA\""
-            }
-        }
+    func customAction() throws{
+        quizzes = quizzesUseCase.getQuizzes()
+        try quizzesUseCase.refreshData()
```

Comment:
If you delete the app and run it, the first time it won't show any quizzes because there are no quizzes yet in Core Data. To fix that you can pass a completion closure to `quizzesUseCase.refreshData()` and call that closure when the new data is saved to CoreData. In the closure you can call `quizzesUseCase.getQuizzes()` and reload the tableView. At the moment the table view is always showing old data (data that was in core data before the new api call).